aws-android-sdk-s3/src/test/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadInputStreamTest.java

```diff
@@ -0,0 +1,193 @@
+        verify(transferUtility, times(1)).upload(eq(DEFAULT_BUCKET), eq(key), any(File.class),
+                any(ObjectMetadata.class), isNull(CannedAccessControlList.class), isNull(TransferListener.class));
+    }
+
+    /**
+     * Test that upload(String key, String inputStream, UploadOptions options) with
+     * all parameters specified calls the File based upload method with the same parameters provided as input.
+     * @throws IOException
+     */
+    @Test
+    public void testUploadWithAllParametersSet() throws IOException {
+        String key = getRandomString();
+        String bucket = getRandomString();
+        transferUtility.upload(key, inputStream, UploadOptions.builder()
+                                                              .bucket(bucket)
+                                                              .objectMetadata(metadata)
+                                                              .cannedAcl(cannedAcl)
+                                                              .transferListener(listener)
+                                                              .build());
+        verify(transferUtility, times(1)).upload(eq(bucket), eq(key), any(File.class),
+                eq(metadata), eq(cannedAcl), eq(listener));
+    }
+
+    /**
+     * Verify that the File does in fact get deleted after the upload is complete.
+     */
+    @Test(expected = AssertionError.class)
+    public void testDeleteTempFileAfterUpload() throws InterruptedException {
+        final CountDownLatch deleteFileLatch = new CountDownLatch(1);
+        final String key = getRandomString();
+        final Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    observer = transferUtility.upload(key, inputStream);
+                    observer.setTransferListener(new TransferListener() {
+                        @Override
+                        public void onStateChanged(int id, TransferState state) {
+                            if (state == TransferState.COMPLETED) {
+                                if (!new File(observer.getAbsoluteFilePath()).exists()) {
+                                    deleteFileLatch.countDown();
+                                }
+                            }
+                        }
+
+                        @Override
+                        public void onProgressChanged(int id, long bytesCurrent, long bytesTotal) {}
+
+                        @Override
+                        public void onError(int id, Exception ex) {}
+                    });
+                } catch (IOException ioException) {
+                    ioException.printStackTrace();
+                }
+            }
+        });
+
+        thread.start();
+
+        if (!deleteFileLatch.await(2, TimeUnit.SECONDS)) {
+            shadowOf(getMainLooper()).idle();
+            fail("File was not deleted after upload.");
+        }
+    }
```

Comment:
While this test does pass now, expecting an `AssertionError` is essentially saying that the thing this test is supposed to be verifying, _doesn't_ work, so I don't think we want to do that. 

I looked into this a bit more and it looks like the upload was not getting started because there are some checks that verify the device has a network connection.  Since this is a unit test, and doesn't run on a device, those checks fail.    The goal of this test is only to verify that the temp file is deleted after upload, so instead of trying to setup a bunch of mocks to get a real upload to happen, I'd suggest just calling manually setting the state to COMPLETED, after calling `upload`, and then verifying that the file doesn't exist any more, like this:

```suggestion
    @Test
    public void testDeleteTempFileAfterUpload() throws IOException {
        // Call upload, so that a temporary File will be created
        final String key = getRandomString();
        observer = transferUtility.upload(key, inputStream);

        // Verify the file exists
        File file = new File(observer.getAbsoluteFilePath());
        assertTrue(file.exists());

        // Set state to COMPLETED
        Context context = InstrumentationRegistry.getInstrumentation().getContext();
        TransferStatusUpdater transferStatusUpdater = TransferStatusUpdater.getInstance(context);
        transferStatusUpdater.updateState(observer.getId(), TransferState.COMPLETED);

        // Verify the file was deleted
        assertFalse(file.exists());
    }
```